### ISSUE TITLE
npm module ending in '.js' breaks typescript AMD compatibility

### ISSUE THREAD

**Issue by grind086** [Microsoft/TypeScript#16444]
**TypeScript Version:**  2.3.4

I've recently come across an issue where having an npm module ending in `.js` breaks typescript's compatibility with requirejs. I install the package via

    npm install pixi.js @types/pixi.js

then import it with something like

```ts
import * as PIXI from 'pixi.js'
```

which compiles to

```js
define("app", ["require", "exports", "pixi.js"], function (require, exports, PIXI) {
    // ...
});
```

Modules ending in `.js` are against the AMD specs ([see here](https://github.com/amdjs/amdjs-api/wiki/AMD#module-id-format-)), and the more practical side of the problem is that requirejs ignores baseUrl and paths configurations for those modules ([documented here](http://requirejs.org/docs/api.html#jsfiles)). 

There are two possible solutions I can think of. First is simply outputting an error when compiling a module ending in `.js` to AMD. This shouldn't break anything, and might spur package maintainers to comply with the specs. Second is forcing compliance by replacing the `.js` with `js` or `_js`. This could break projects for a small number of people, but the fix would be simply updating/adding a path configuration to their module loader.

**Comment by grind086** [comment]
This isn't a part of this issue, but I tried getting around this using typescript's path configuration

```json
{
    "baseUrl": ".",
    "paths": {
        "PIXI": ["node_modules/pixi.js/dist/pixi.js"]
    }
}
```

```ts
import * as PIXI from 'PIXI';
```

however that breaks the association with the `@types` module. Adding triple-slash directive doesn't fix this, because the module name in the typings package no longer matches the import. See #16437.

**Comment by grind086** [comment]
I'm noticing this only seems to be an issue when importing a module via `node_modules` with an ambient declaration. For example, in [this repo](https://github.com/grind086/ts-amd/tree/8fc5f6c42d9ba8865c9f428b2757fa1b3ce0ee8e) the compiler will give a `TS2307: Cannot find module './module.js'` (though it will still compile module.js to an invalid amd). If we move the module into `node_modules` and add an ambient declaration like [this](https://github.com/grind086/ts-amd/tree/d11917220927321995700f52077a9e49fed32507), it compiles with no errors. If we remove the declaration it will give the error `TS2307: Cannot find module 'module.js'`

To sum up, the compiler doesn't recognize that the following shouldn't be a valid module declaration

    declare module 'module.js' {
        /* ... */
    }

    import * as module from 'module.js'

**Comment by mhegazy** [comment]
The compiler does not rewrite module names. the text you have in the string literal will be used in the output verbatim. you can instead define a module "pixi_js" in a .d.ts file:

```ts
// globals.d.ts
declare module "pixi_js" {
    import * as PIXI from "pixi.js";
    export = PIXI;
}

```

and then use it as:

```ts
import * as PIXI from 'pixi_js'
```

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
